text_file_to_ui/text_file_to_ui/UserParser.cs

```diff
@@ -12,25 +11,40 @@ public User UserParse(Reader a)
             a = userFileReader.FileOpen();
             if (a != null)
             {
-                b.firstName = a.firstNameString.Substring(4);
-                b.lastName = a.lastNameString.Substring(8);
-                b.yearOfBirth = a.yearOfBirthString.Substring(15);
-                b.city = a.cityString.Substring(13);
-                b.faculty = a.facultyString.Substring(9);
-                b.role = a.roleString.Substring(6);
-                if (b.role == "student")
+#if DEBUG
+                try
+#endif
```

Comment:
Uh, uh. Ovo izbjegavamo pod svaku cijenu, osim recimo kada iz različitih direktorija čitamo neke konfiguracije u Debug i Release načinu rada. Dakle, ovo mora van, a softver se s neispravnim datotekama treba testirati i u jednom i u drugom načinu rada (gore se mijenjaju kod zelenog "Play" gumba).

Inače, za `try` block bi bilo najbolje da obuhvaća samo onu jednu liniju gdje očekujemo da će program puknuti. Npr., u ovom slučaju možemo dodati još jednu klasu UserFileValidator koja baci exception "UserFileInvalidException" ako u datoteci ne postoje svi traženi atributi ("Ime", "Prezime" itd.).
Vlastite Exceptione stvaramo kao najnormalnije klase, ali da nasljeđuju Exception klasu:
(`class UserFileInvalidException : Exception {...`). Probaj to napraviti samostalno.